### PR TITLE
chore: move all instances of dayjs to date.ts

### PR DESCRIPTION
**Pull request by Prashanth-Nelli**

### CODE REVIEW

File: src/components/IncidentDetails/index.tsx [hunk 1]
```diff
@@ -73,23 +73,34 @@ export const IncidentDetails = ({
...
+  };
 
   const watchCreatedAt = watch("created_at");
   const watchType = watch("type");
   const watchPriority = watch("priority");
   const watchCommanders = watch("commanders");
 
   const formattedCreatedAt = useMemo(
-    () => formatDate({ date: watchCreatedAt }),
+    () => formatCreatedAt(watchCreatedAt, "DATE"),
```
Comment (moshloop) [inline src/components/IncidentDetails/index.tsx:99]: Why do we need these - can't we use relativeDateTime and duration directly ?

File: src/utils/date.ts [hunk 2]
```diff
@@ -10,4 +37,125 @@ export function isDate(dateString?: ValueType) {
...
+ */
+export const dateToJSDate = (value: DateConfig) => dayjs(value).toDate();
+
+/**
+ *
+ * @param value date to format
+ * @param format custom string format for the date
+ * @returns formatted string of the date
+ */
+export const customDateFormattor = (value: DateConfig, format: string) =>
```
Comment (moshloop) [inline src/utils/date.ts:136]: Why do we need this ?

File: src/components/SchemaResourcePage/SchemaResourceList.tsx [hunk 3]
```diff
@@ -82,8 +82,12 @@ function SchemaResourceListItem({
       <Cell className="shrink-0">
         {!!source && <a href={`${source}`}>Link</a>}
       </Cell>
-      <Cell className="text-gray-500">{dayjs(created_at).fromNow()}</Cell>
-      <Cell className="text-gray-500">{dayjs(updated_at).fromNow()}</Cell>
+      <Cell className="text-gray-500">
+        {relativeDateTime(created_at, true)}
```
Comment (moshloop) [inline src/components/SchemaResourcePage/SchemaResourceList.tsx:86]: Why do we need a `true` param? Rather create a new function

File: src/components/TimeRangePicker/RecentlyRanges.tsx [hunk 4]
```diff
@@ -23,12 +22,16 @@ const RecentlyRangesFC = ({
         <button
           type="button"
           onClick={() => applyTimeRange(range)}
-          key={`${dayjs(range.from).format()}${dayjs(range.to).format()}`}
+          key={`${formatDate(range.from, {
```
Comment (moshloop) [inline src/components/TimeRangePicker/RecentlyRanges.tsx:25]: Rather use `formatLongDate` and `formatTimeRange`

File: src/components/TimeRangePicker/TimeRangePickerBody.tsx [hunk 5]
```diff
@@ -55,8 +62,8 @@ const TimeRangePickerBodyFC = ({
       if (
         !recentRanges.find(
           (el: RangeOption) =>
-            dayjs(el.from).toISOString() === dayjs(range.from).toISOString() &&
-            dayjs(el.to).toISOString() === dayjs(range.to).toISOString()
+            dateToISOString(el.from) === dateToISOString(range.from) &&
```
Comment (moshloop) [inline src/components/TimeRangePicker/TimeRangePickerBody.tsx:65]: rename to `formatISODate`

File: src/utils/date.ts [hunk 6]
```diff
@@ -10,4 +37,125 @@ export function isDate(dateString?: ValueType) {
...
+}
+
+export enum DATE_FORMATS {
+  TIME = "h:mm A",
+  SHORT = "MMM DD",
+  LONG = "h:mm A, MMM DD, YYYY",
+  TIME_RANGE = "YYYY-MM-DD HH:mm"
+}
+
+type formatDateOptions = {
```
Comment (moshloop) [inline src/utils/date.ts:49]: These options are not intuitive, rather create dedicated functions

File: src/utils/date.ts (hunk 2 above)
Comment (Prashanth-Nelli) [reply src/utils/date.ts:140]: We have created TimeRangePicker component which similar to Grafana time range picker this function is being used in that component & Canary Status Chart xAxis value formatting.

File: src/utils/date.ts (hunk 6 above)
Comment (Prashanth-Nelli) [reply src/utils/date.ts:49]: created dedicated functions as suggested.

Comment (moshloop) [reply src/utils/date.ts:49]: Don't export the formatDate

Comment (Prashanth-Nelli) [reply src/utils/date.ts:49]: Done

File: src/utils/date.ts (hunk 2 above)
Comment (moshloop) [reply src/utils/date.ts:139]: Then it should in a formatFunction, we don't want to distribute date formats throughout the code base

Comment (Prashanth-Nelli) [reply src/utils/date.ts]: Done

**Pull request merged**